pkg/iter/iterator.go

```diff
@@ -608,7 +613,12 @@ func (i *entryIteratorForward) Entry() logproto.Entry {
 	return i.cur.entry
 }
 
-func (i *entryIteratorForward) Close() error { return nil }
+func (i *entryIteratorForward) Close() error {
+	if !i.loaded {
```

Comment:
I think the `entryIteratorForward` and `entryIteratorBackward` could be more easily expressed as `reverseIterator(iter, limit)` with 0 limit being unlimited, but that's unrelated.

Reply:
don't think we will support unlimited. but for sure those could refactored into a single one with direction. all yours.